### ISSUE TITLE
Add a `Meta.allTables` extension variable/function

### ISSUE THREAD

**Issue by donard-commedagh** [komapper/komapper#681]
### Use case

It would be nice (when developing/testing) to perform CREATE and/or DROP on all tables at once. When there a few tables it is easy to list them all and also not forget one. As the table count grows, there is an increased likelihood of forgetting to include one when creating the list manually. _(I'm currently testing against a DB with ~40 tables.)_

### Describe the solution you'd like

Instead of running
```kotlin
val query: Query<Unit> = QueryDsl.create(
    Meta.alpha,
    Meta.bravo,
    Meta.charlie,
    Meta.delta,
    Meta.echo,
    Meta.foxtrot,
    ..
)
```

Run
```kotlin
val query: Query<Unit> = QueryDsl.create(Meta.allTables)
```
_Note:_ This might need to be named something else or located somewhere else in the unlikely event that a database has a table called `AllTables`

### Versions

- Komapper: 1.1.2
- Kotlin: 1.7.10
- JVM: Corretto 14
- Database: SQL Server and H2 In-Memory
- JDBC Driver or R2DBC Driver: Dependencies as pulled in via `komapper-dialect-h2-r2dbc` and `komapper-dialect-sqlserver-r2dbc`
- OS: Windows 10

**Comment by nakamura-to** [comment]
It isn't easy.

If we generate `Meta.allTables` at compile-time, we will not benefit from [KSP's incremental processing](https://kotlinlang.org/docs/ksp-incremental.html). Because we always have to process all entities and entity definitions to know all meta names.

Do you have any good ideas?

**Comment by donard-commedagh** [comment]
After a bit of reading into KSP, which I still have limited knowledge of, I can't see a way to do it without requiring a reprocessing of all entities and definitions.

The only suggestion I would have is in opt-in build flag, which would generate the `allTables` variable at the cost of a full reprocessing.

This is be no means a must-have request, it would just be a convenience. I'm quite happy for this to closed as 'won't implement' if you want.

**Comment by nakamura-to** [comment]
Thanks for sharing your thoughts. 
I won't fix it at this time.

**Comment by dave08** [comment]
Maybe have an extension property generated at compile time that uses reflection at run-time (basically just used for unit tests... so maybe not so bad to have that extra overhead?)

**Comment by nakamura-to** [comment]
It is not possible to collect all extended properties in reflection.
Why do we need a sealed interface/class? I am not sure if it will help in this case.

There is a way to enable the Metamode listing with the KSP option as follows:

```kotlin
ksp {
    arg("komapper.enableMetamodelListing", "true")
}
```

When this option is enabled, Komapper always processes all entity classes. This approach will result in some loss of compilation performance. 
However, if you want we would provide it.

**Comment by dave08** [comment]
Say you'd just Add a special `sealed interface EntityMetaBaseForMeta1` and derive all `Meta1` entity meta classes from it, we could retrieve a list of all those classes by doing something like this in the test code `val entities: List<KClass<*>> = EntityMetaBaseForMeta1::class.sealedSubclasses`, then you wouldn't need the extra pass in KSP...

**Comment by nakamura-to** [comment]
Who should provide EntityMetaBaseForMeta1?

Kotlin [documentation](https://kotlinlang.org/docs/sealed-classes.html) says:

> All direct subclasses of a sealed class are known at compile time. No other subclasses may appear outside a module within which the sealed class is defined. 

So Komapper cannot provide EntityMetaBaseForMeta1.

**Comment by nakamura-to** [comment]
This issue has been fixed by #857.
